docs/targets/polkadot.rst

```diff
@@ -62,3 +61,42 @@ The following example shows how call flags can be used:
 .. include:: ../examples/polkadot/call_flags.sol
   :code: solidity
 
+
+Reverts and error data decoding
+_______________________________
+
+When a contract reverts, the returned error data is what
+`EVM would return <https://docs.soliditylang.org/en/v0.8.20/control-structures.html#panic-via-assert-and-error-via-require>`_.
+``assert()``, ``require()``, or ``revert()`` will revert the contract execution, where the revert reason 
+is supplied as the contracts output (if any). Solidity contracts can also revert with a `Panic`.
+Uncaught exceptions from calling and instantiating contracts or transferring funds will be bubbled 
+up back to the caller.
+
+The metadata contains all error variants that the contract knows about in the ``lang_error`` field.
+The 4 bytes `selector` of the error data can be seen as the enum discriminator. However, because 
+SCALE encoding does not allow discriminators larger than 1 byte, the hex-encoded error selector 
+is provided as the enum variant name in the metadata (the selector could also be calculated by
+reconstructing and hashing the error signature based on the enum variant types).
+
+.. note::
+
+    Solidity knows about `Error`, `Panic` and 
+    `custom errors <https://docs.soliditylang.org/en/latest/abi-spec.html#errors>`_.
+    Solang does not yet support custom errors. For now, only `Error` (selector of `0x08c379a0`) 
+    and `Panic` (selector of `4e487b71`) will be returned and occur in the metadata.
+
+The general process of decoding the output data for Solang Solidity contracts is as follows:
+
+1. If the revert flag is **not** set, the contract didn't revert and the output should be encoded as specified in the message spec.
+2. The compiler version must be solang > 0.3.1, or the error data can't be decoded (check the ``compiler`` field in the contract metadata).
+3. If the output length is smaller than 4 bytes, the error data can't be decoded (contracts may return empty error data, for example if ``revert()`` without arguments is used).
+4. If the first 4 bytes of the output do **not** match any of the selectors found in ``lang_error``, the error can't be decoded.
+5. **Skip** the selector (first 4 bytes) and decode the remaining data according to the corresponding variant type found in `lang_error`.
```

Comment:
Some example code would be great

Reply:
What code example do you have in mind?

I don't see what a code example would add. Also, a Typescript or Rust snippet to do this will be quite long I don't see why this belongs into our docs, a solidity example is uselesse because try-catch should be used anyways.

Reply:
How about a link to an integration test?

Reply:
Well, in the integration test we always just match the output directly against the expected output bytes (integration tests are not examples...)